### DISCUSSION TITLE
Schedule each x ticks until condition is met

### DISCUSSION THREAD

**Discussion by KyllianGamer** [Minestom/Minestom#1499] [Help] [answered]
How would I use the scheduler on a player to schedule a task that runs each X ticks/seconds/... until some condition is met? Right now for running a task each x ticks for x ticks I just use 2 tasks, one that repeats and one that's just a delay and cancels the other one. Is this bad practice? If so, how else should I do it? And what about non-time-related conditions?

**Comment by HSGamer** [comment, marked as answer]
You can use [`submitTask`](https://javadoc.minestom.net/net/minestom/server/timer/Scheduler.html#submitTask(java.util.function.Supplier,net.minestom.server.timer.ExecutionType)) that supplies [`TaskSchedule`](https://javadoc.minestom.net/net/minestom/server/timer/TaskSchedule.html)
```java
scheduler().submitTask(() -> {
  if (stopCondition) {
   return TaskSchedule.stop();
  } else {
    // your logic
    return TaskScheduke.tick(20); // Delay
  }
})
```

**Reply by KyllianGamer** [reply]
Thanks!
